src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -117,12 +109,14 @@ public void run() {
           if (responseCode >= 400) {
             LOG.warn("Http Error Code: {} for {}, \nReason: {}", responseCode, mUrl.toString(),
                     response.getStatusLine().getReasonPhrase());
-            mAccount.authenticate();
-            request.removeHeaders("X-Auth-Token");
-            request.addHeader("X-Auth-Token", mAccount.getAuthToken());
-            LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
-            response = client.execute(request);
-            responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode == 401) { // Unauthorized error
+              mAccount.authenticate();
+              request.removeHeaders("X-Auth-Token");
+              request.addHeader("X-Auth-Token", mAccount.getAuthToken());
+              LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
+              response = client.execute(request);
+              responseCode = response.getStatusLine().getStatusCode();
+            }
```

Comment:
@djalova So what should we do if it's not 401? I guess we need to throw exception?